Dryad Digital Repository.js

```diff
@@ -0,0 +1,715 @@
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	// Dryad search page
+	if (url.match('^https?://(?:www\\.)?datadryad\\.org/discover')) {
+		return 'multiple';
+	} else {
+		var result = ZU.xpathText(doc,'//meta[@name="DC.type"][1]/@content');
+		// Dryad data package
+		if (result === 'Article') {
+			return 'journalArticle';
+		// Dryad data file
+		} else if (result === 'Dataset') {
+			//return 'dataset';
+			return 'journalArticle';
+		}
+	}
+	return false;
+}
+
+var packageXpath    = '//meta[@name="DCTERMS.isPartOf"]/@content';
+var multiplesXpath  = '//li[contains(@class,"ds-artifact-item")]/div/a/@href';
+var doiXpath        = '//meta[@name="DC.identifier"][starts-with(@content,"doi:10.5061/dryad.")]/@content';
+var doiUrl          = 'http://dx.doi.org/';
+
+function doWeb(doc, url) {
+	var itemType = detectWeb(doc, url);
+	if (itemType === 'journalArticle') {
+		var translator = Zotero.loadTranslator('import');
+		// use the Embedded Metadata translator
+		translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+		// note: this handler function is called for a data package
+		// or a data file. If called for a data package as part of setting
+		// the seeAlso value, then the 'doc' variable is still aliasing
+		// the data file's DOM
+		translator.setHandler('itemDone', function (obj, item) {
+			var itemUrl = item.url;  // scoped to this handler
+			var doi = itemUrl.replace(/^.*?doi:/, '');
+			item.DOI = doi;
+			item.itemId = doi;       // internal value for seeAlso relation
+			item.url = doiUrl + doi;
+			item.itemType = itemType;
+			// add a seeAlso value if this is a data file 'doc' is
+			// scoped to the outer doWeb call, and seeAlso is only set
+			var pdoi = ZU.xpathText(doc, packageXpath);
+			var docdoi = doc.location.href.replace(/^.*?doi:/, '');
+			if (pdoi && pdoi !== doi) {
+				item.seeAlso = [doiUrl + pdoi];  // dx.doi.org url for seeAlso
+			// import called on data file, and we are in the callback for
+			// the data package being associated with it
+			} else if (docdoi !== doi) {
+				item.seeAlso = [doiUrl + docdoi];
+			// import called on data package
+			} else {
+				item.seeAlso = [];
+			}
+			// signal a dataset until supported per
+			// https://www.zotero.org/support/dev/translators/datasets
+			item.extra = '{:itemType: dataset}';
+			item.archive = 'Dryad Digital Repository';
+			var upd = [ "reportType", "letterType", "manuscriptType",
+				"mapType", "thesisType", "websiteType",
+				"presentationType", "postType", "audioFileType"];
+			for (var i = 0; i < upd.length; i++) item[upd[i]] = itemType;
+			item.attachments = [];
+			item.shortTitle = '';
+			item.complete();
+		});
+		translator.getTranslatorObject(function (obj) {
+			obj.doWeb(doc, url);
+		});
+		var package_doi = ZU.xpathText(doc, packageXpath);
+		if (package_doi) {
+			package_doi = package_doi.substring(4); // omit leading 'doi:'
+			ZU.processDocuments([doiUrl + package_doi], function (d) {
```

Comment:
Your callback should just be `doWeb`, since `ZU.processDocuments` passes back document and url as parameters.

nit: the URL can be a string. Doesn't have to be an array.